src/mlpack/tests/knn_test.cpp

```diff
@@ -1382,6 +1410,33 @@ BOOST_AUTO_TEST_CASE(MoveConstructorRTreeTest)
 }
 
 
+/**
+ * Test the move constructor using Binary Space trees.
+ */
+BOOST_AUTO_TEST_CASE(MoveConstructorBinarySpaceTreeTest)
+{
+  arma::mat dataset = arma::randu<arma::mat>(5, 500);
+  typedef NeighborSearch<NearestNeighborSort, EuclideanDistance, arma::mat,
+      KDTree> NeighborSearchType;
+  NeighborSearchType* knn = new NeighborSearchType(std::move(dataset));
+
+  // Get predictions.
+  arma::mat distances, distances2;
+  arma::Mat<size_t> neighbors, neighbors2;
+
+  knn->Search(3, neighbors, distances);
+
+  // Use move constructor.
+  NeighborSearchType knn2(std::move(*knn));
```

Comment:
It looks like we are only verifying `move constructor`. However, we should also check `move assignment`. To do that below is the code (Add it after knn2.search())
```
NeighborSearchType knn3 = std::move(*knn);
knn3.search(neighbors3, distances3);
``` 

It looks like this also got missed in #1934. So it would be great if you modify that test case as well.
For more reference:
https://en.cppreference.com/w/cpp/language/move_assignment